### PR TITLE
add async in read from s3

### PR DESCRIPTION
**Pull request by YLShiJustFly**
Signed-off-by: YLShi <344515349@qq.com>

### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by wuhongsong** [comment]
recheck

File: curvefs/src/client/s3/client_s3_cache_manager.cpp [hunk 1]
```diff
@@ -532,14 +532,35 @@ int FileCacheManager::ReadFromS3(const std::vector<S3ReadRequest> &requests,
             uint64_t start = butil::cpuwide_time_us();
             if (async) {
```
Comment (wuhongsong) [inline curvefs/src/client/s3/client_s3_cache_manager.cpp:533]: if async is enable, you can set（bool async = true;)

**Comment by cw123** [comment]
ReadFromS3的实现中，还是默认的sync read。你的修改是解决 #2070 这个问题的吗？

In function of ReadFromS3, the default is still sync read. 
Does your change address #2070?

**Comment by wuhongsong** [comment]
recheck

**Review by cw123 — COMMENTED** [review]
1. most read logic of sync read between async are same,  extract common logic
2.  add some unit test case

**Comment by cw123** [comment]
recheck

**Comment by cw123** [comment]
recheck

**Comment by ilixiaocui** [comment]
This pr has been opened for a long time, and the review comments have not been replied. Please resubmit the pr if necessary.

**Pull request closed without merging**